### DISCUSSION TITLE
.get_me() method not working on Client initalized via OAuth 2.0 Authorization Code Flow with PKCE (User Context)

### DISCUSSION THREAD

**Discussion by c-romano** [tweepy/tweepy#2006] [Questions] [answered]
The .get_me() method on a Client object initialized with the OAuth 2.0 User Context flow is giving a TypeError, "Consumer key must be string or bytes, not NoneType". Don't think this should be happening as the flow doesn't require a Consumer Key on the Client object based on the docs. Seems that callback URL is working as it should be. Maybe I'm missing something though.

I created the Client object in the same way as the docs, aka:
```
access_token = oauth2_user_handler.fetch_token(
    "Authorization Response URL here"
)

client = tweepy.Client("Access Token here")
```

Tried the .get_user() method on the object while giving it the auth'd user's username as an extra test. This gives a 401 unauthorized error though.

**Comment by Harmon758** [comment]
What exactly are you trying?

> Tried the .get_user() method on the object while giving it the auth'd user's username as an extra test. This gives a 401 unauthorized error though.

Again, what exactly are you trying?
What OAuth 2.0 scopes did you authenticate with?

In the future, please provide the full traceback.

**Comment by c-romano** [comment]
Thanks for the explanation, and will do. I used the scopes follows.read, users.read, tweets.read, and offline.access. 

This handler is defined in views.py:
```
oauth2_user_handler = tweepy.OAuth2UserHandler(
    client_id=os.getenv('CLIENT_ID'),
    redirect_uri="http://127.0.0.1:8000/init/callback",
    scope=["follows.read", "users.read", "tweet.read", "offline.access"],
    # Client Secret is only necessary if using a confidential client
    client_secret=os.getenv('CLIENT_SECRET')
    )
```

And then I'm using Django, so I have this auth view:
```
def oauth2(request):

    return HttpResponseRedirect(oauth2_user_handler.get_authorization_url())
```
^^ this works to get to Twitter for auth.

And then this returns to the callback view, which creates Client via the OAuth 2.0 PKCE User flow via the line of code in the original post. The full code snippet is here:

```
def callback(request):

    token = oauth2_user_handler.fetch_token(request.build_absolute_uri())

    user_client = tweepy.Client(
        token
    )

    me = user_client.get_me()
```

The full traceback from the Django debugger is:
```
Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Python310\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\***\twitterInit\views.py", line 45, in callback
    me = user_client.get_me()
  File "C:\Python310\lib\site-packages\tweepy\client.py", line 2517, in get_me
    return self._make_request(
  File "C:\Python310\lib\site-packages\tweepy\client.py", line 126, in _make_request
    response = self.request(method, route, params=request_params,
  File "C:\Python310\lib\site-packages\tweepy\client.py", line 68, in request
    auth = OAuth1UserHandler(
  File "C:\Python310\lib\site-packages\tweepy\auth.py", line 31, in __init__
    raise TypeError("Consumer key must be string or bytes, not "

Exception Type: TypeError at /init/callback
Exception Value: Consumer key must be string or bytes, not NoneType

```

Maybe I'm just missing something?

**Comment by Harmon758** [comment, marked as answer]
[`Client.get_me`](https://tweepy.readthedocs.io/en/v4.12.0/client.html#tweepy.Client.get_me) defaults to using OAuth 1.0a User Context to authenticate. You need to pass `user_auth=False`.

As for the 401 error, `token` is a dictionary with the response data from fetching the token.
You need to pass the actual access token as a string, i.e. `token["access_token"]`, when initializing `Client`.

> I used the scopes follows.read, users.read, tweets.read, and offline.access.

Note, you use `tweet.read`, which is correct.

**Reply by c-romano** [reply]
Ahh I see, my mistake. That did fix it. Thanks for the explanation.
